apps/coscrad-frontend/src/components/resources/bibliographic-references/bibliographic-reference-detail.presenter.tsx

```diff
@@ -0,0 +1,16 @@
+import { IBibliographicReferenceViewModel, IDetailQueryResult } from '@coscrad/api-interfaces';
+import { buildBibliographicReferenceDetailPresenter } from './build-bibliographic-reference-detail-presenter';
+
+export const BibliographicReferenceDetailPresenter = (
+    detailQueryResult: IDetailQueryResult<IBibliographicReferenceViewModel>
+): JSX.Element => {
+    const {
+        data: {
+            data: { type: bibliographicReferenceType },
```

Comment:
Yes- this is a modelling \ design smell. Very good point.

One refactor I'd like to do is to make it so that the detail response is simply an `IBaseViewModel` with `actions` as part of the `IBaseViewModel`. This may point to places where we've been using `IBaseViewModel` in place of a `HasId`, but otherwise should be a simple refactor.

As for the `IIndexQueryResponse`, I think it is ok to leave the data key as we wrap it behind the `ILoadable` abstraction and the custom hooks in the Redux layer.